### PR TITLE
Add electra DB

### PR DESCRIPTION
**Pull request by terencechain**
This PR adds electra changes to db. Mainly two things and a note
1.) Add electra block support
2.) Add electra state support
3.) Refactor `saveStatesEfficientInternal` into multiple helpers. The functionality remains the same. It needs to be done to reduce cognitive complexity or CI will be red

### CODE REVIEW

File: beacon-chain/db/kv/state.go [hunk 1]
```diff
@@ -229,160 +229,142 @@ func (s *Store) saveStatesEfficientInternal(ctx context.Context, tx *bolt.Tx, bl
...
+		return err
+	}
+	pbState.Validators = valEntries
+	if err := valIdxBkt.Put(rootHash, validatorKey); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Store) processEPBS(ctx context.Context, pbState *ethpb.BeaconStateElectra, rootHash []byte, bucket, valIdxBkt *bolt.Bucket, validatorKey []byte) error {
```
Comment (james-prysm) [inline beacon-chain/db/kv/state.go:352]: should this be called EPBS instead of electra? is there a TODO to rename this?

Comment (terencechain) [reply beacon-chain/db/kv/state.go:352]: sorry this was a typo, I'll fix it

Comment (terencechain) [reply beacon-chain/db/kv/state.go]: c217712c06a81b6925183e244ff922ad49341aa9

**Review by james-prysm — APPROVED** [review]
LGTM

File: beacon-chain/db/kv/key.go [hunk 2]
```diff
@@ -51,3 +51,17 @@ func hasDenebBlindKey(enc []byte) bool {
 	}
 	return bytes.Equal(enc[:len(denebBlindKey)], denebBlindKey)
 }
+
+func hasElectraKey(enc []byte) bool {
+	if len(electraKey) >= len(enc) {
```
Comment (kasey) [inline beacon-chain/db/kv/key.go:56]: We don't really need to do this, it would be fine to just call bytes.Equal.

**Review by kasey — DISMISSED** [review]

File: beacon-chain/db/kv/key.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/db/kv/key.go:56]: @kasey it is required or a runtime error for `slice bounds out of range` will happen when `len(enc)` is less than `len(electraKey)`

File: beacon-chain/db/kv/state_test.go [hunk 3]
```diff
@@ -155,7 +182,7 @@ func TestState_CanSaveRetrieve(t *testing.T) {
 				savedSt, err := db.State(context.Background(), root)
 				require.NoError(t, err)
 
-				assert.DeepSSZEqual(t, st.ToProtoUnsafe(), savedSt.ToProtoUnsafe())
+				require.DeepEqual(t, st.ToProtoUnsafe(), savedSt.ToProtoUnsafe())
```
Comment (prestonvanloon) [inline beacon-chain/db/kv/state_test.go:185]: Why is this changed?

File: beacon-chain/db/kv/key.go (hunk 2 above)
Comment (prestonvanloon) [reply beacon-chain/db/kv/key.go:56]: You are thinking of bytes.Prefix, i think. It would be safe to call bytes.HasPrefix without this check

https://pkg.go.dev/bytes?utm_source=godoc#HasPrefix
```go
func hasElectraKey(enc []byte) bool {
	return bytes.hasPrefix(enc, electraKey)
}
```

Comment (kasey) [reply beacon-chain/db/kv/key.go:56]: You're right, this is basically replicating `bytes.HasPrefix`.

File: beacon-chain/db/kv/state_test.go (hunk 3 above)
Comment (terencechain) [reply beacon-chain/db/kv/state_test.go:185]: I fixed the library but forget to revert back the line. I'll fix it

**Review by james-prysm — APPROVED** [review]

**Review by prestonvanloon — APPROVED** [review]
